### PR TITLE
APP-1387 Add switch e2e tests

### PR DESCRIPTION
**Pull request by ehhong**
### Notes
- Added helper function for custom events with parameters (should merge https://github.com/viamrobotics/prime/pull/194 first)
- Moved `hexToRGB()` function into helpers
- Updated stories to include `labelposition` attribute
- Fixed disabled/readonly styling in `switch` component, verified with @nathartman 

_designs_
image

_updated storybook_
image
image

Test | Behavior
-- | --
Variant: annotated on | GIVEN a "variant" attribute of "annotated" has been applied to the v-switch elementAND the "value" attribute is "on" WHEN the element is renderedTHEN the switch should display in the on positionAND the text of "ON" should display to the right of the switch element
Variant: annotated off | GIVEN a "variant" attribute of "annotated" has been applied to the v-switch elementAND the "value" attribute is "off" WHEN the element is renderedTHEN the switch should display in the off positionAND the text of "OFF" should display to the right of the switch element
Variant: default or no variant specified | GIVEN no "variant" attribute has been applied to the v-switch elementWHEN the element is renderedTHEN the switch should display in the off position
Click/Value: start on | GIVEN a "value" attribute has been applied to the v-switch elementAND the "value" attribute value is "on" WHEN the element is renderedAND the element is clickedTHEN the switch should display in the off positionAND the switch should emit an "input" event
Click/Value: start off | GIVEN a "value" attribute has been applied to the v-switch elementAND the "value" attribute value is "off" WHEN the element is renderedAND the element is clickedTHEN the switch should display in the on positionAND the switch should emit an "input" event
Keydown Enter/Value: on | GIVEN a "value" attribute has been applied to the v-switch elementAND the "value" attribute value is "on" WHEN the element is renderedAND the element is clickedTHEN the switch should display in the off positionAND the switch should emit an "input" event
Keydown Enter/Value: off | GIVEN a "value" attribute has been applied to the v-switch elementAND the "value" attribute value is "off" WHEN the element is renderedAND the element is clickedTHEN the switch should display in the on positionAND the switch should emit an "input" event
Label: default | GIVEN a "label" attribute has been applied to the v-switch elementWHEN the element is renderedTHEN the label attribute text should display above the switch element
Disabled | GIVEN a "disabled" attribute has been applied to the v-switch elementAND the "disabled" attribute is "true"WHEN the element is renderedTHEN the switch should render as disabled
Readonly | GIVEN a "readonly" attribute of "true" has been applied to the v-switch elementWHEN the element is renderedTHEN the switch should render as readonly
Label Position: left | GIVEN a "labelposition" attribute of "left" has been applied to the v-switch elementAND a "label" attribute has been applied to the v-switch element with some textWHEN the element is renderedTHEN the label text should display to the left of the switch element
Label Position: top | GIVEN a "labelposition" attribute of "top" has been applied to the v-switch elementAND a "label" attribute has been applied to the v-switch element with some textWHEN the element is renderedTHEN the label text should display to the top of the switch element
Label Position: default | GIVEN no "labelposition" attribute of "top" has been applied to the v-switch elementAND a "label" attribute has been applied to the v-switch element with some textWHEN the element is renderedTHEN the label text should display to the top of the switch element
Tooltip: label | GIVEN a tooltip attribute has been applied to a switch elementAND a label attribute has been applied to a switch elementWHEN the element is renderedTHEN the label attribute value should render above the switch elementAND an info icon should be rendered to the right of the label attribute valueAND if the info icon is hoveredTHEN the tooltip attribute value should render above the info icon

### CODE REVIEW

File: tests/lib/helper.ts [hunk 1]
```diff
@@ -1,15 +1,56 @@
 import { expect,Page } from '@playwright/test';
+import config from '../../tailwind.config.cjs'
+
+export const hexToRGB = (color?: string) => {
```
Comment (anjinai) [inline tests/lib/helper.ts:4]: Thanks for moving this over!

**Review by anjinai — APPROVED** [review]
LGTM!!!

File: tests/lib/helper.ts (hunk 1 above)
Comment (Kschappacher) [reply tests/lib/helper.ts:4]: +1

File: playground/switch-test.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="en">
```
Comment (Kschappacher) [inline playground/switch-test.html:2]: Good stuff splitting out the tests! One nit, The src folder is looking a bit full. We might want to separate the .ts .vue and .html files into folders for better organization at some point.

File: tests/lib/helper.ts [hunk 3]
```diff
@@ -1,15 +1,56 @@
...
+    new Promise((resolve) => {setTimeout(resolve, 500, "timeout")}),
+  ])
 
-    expect(value).toBe("timeout")
+  expect(value).toBe("timeout")
 }
 
+// This function adds an event listener for eventName and puts the value of paramName on the window
+// Retrieve the value of paramName by calling await page.evaluate('window.<paramName>')
+export const waitForCustomEventWithParam = (page:Page, eventName:string, paramName:string) => {
```
Comment (Kschappacher) [inline tests/lib/helper.ts:45]: good stuff getting this to work. Are there any concerns with placing the value on the window causing issues between tests? is there any clean up we could do of the value afterwards so another test doesnt pick it up?

**Review by Kschappacher — APPROVED** [review]
One question about the custom events value but otherwise it looks good to me!

File: tests/lib/helper.ts (hunk 3 above)
Comment (Kschappacher) [reply tests/lib/helper.ts:45]: Moving this comment to the other pr

**Review by DTCurrie — APPROVED** [review]
✅

File: playground/switch-test.html (hunk 2 above)
Comment (ehhong) [reply playground/switch-test.html:2]: will fix this in the next e2e test PR!

**Pull request merged**